### ISSUE TITLE
could you provide a pytorch implementation?

### ISSUE THREAD

**Issue by DanceInDark** [idiap/attention-sampling#1]

**Comment by angeloskath** [comment]
Yes but firstly I will focus on completing the keras/tf release and making a pip installable package from PyPI.

If anybody feels like contributing a Pytorch implementation that would be ideal ;-) ...

**Comment by timqqt** [comment]
I just finished the Pytorch implementation. Before releasing, could you help me check the correction and let me know how to cite your work.

**Comment by angeloskath** [comment]
That sounds awesome!

I am a bit swamped at the moment but I can fully help mid-February. Is that too late for you? If yes I can try to help now but I can't promise much.

Thank you so much for your work!

Cheers,
Angelos

**Comment by angeloskath** [comment]
@timqqt Let me know if you need help with anything.

**Comment by timqqt** [comment]
> @timqqt Let me know if you need help with anything.

Thanks. Sorry for the late response. I made some modifications of the Pytorch implementation for my own research. I will try to separate them and I will let you know when I am finished.

**Comment by muzhibenying** [comment]
> > @timqqt Let me know if you need help with anything.
> 
> Thanks. Sorry for the late response. I made some modifications of the Pytorch implementation for my own research. I will try to separate them and I will let you know when I am finished.

Hello, I am a new learner of deep learning and now I'm struggling for a breast cancer project. Because of the lack of ability, I can not convert the keras version implementation to a pytorch version. Are you willing to share your pytorch implementation? Thank you very much.

**Comment by muzhibenying** [comment]
> I just finished the Pytorch implementation. Before releasing, could you help me check the correction and let me know how to cite your work.

Hello, I am a new learner of deep learning and now I'm struggling for a breast cancer project. Because of the lack of ability, I can not convert the keras version implementation to a pytorch version. Are you willing to share your pytorch implementation? Thank you very much.

**Comment by jorism1993** [comment]
I have made a pytorch version and published it here: https://github.com/sara-nl/attention-sampling-pytorch. I have basically rewritten every function to pytorch and tried to adhere as much as possible to the structure in this repo. I have tested it on the speedlimits detection and mega mnist challenges, yielding comparable results to this repository.

**Comment by muzhibenying** [comment]
Thank you very much! It's vey kind of you.



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "jorism1993"<notifications@github.com&gt;; 
发送时间: 2020年5月7日(星期四) 晚上10:10
收件人: "idiap/attention-sampling"<attention-sampling@noreply.github.com&gt;; 
抄送: "左志群"<2590860900@qq.com&gt;; "Comment"<comment@noreply.github.com&gt;; 
主题: Re: [idiap/attention-sampling] could you provide a pytorch implementation? (#1)





 
I have made a pytorch version and published it here: https://github.com/sara-nl/attention-sampling-pytorch. I have basically rewritten every function to pytorch and tried to adhere as much as possible to the structure in this repo. I have tested it on the speedlimits detection and mega mnist challenges, yielding comparable results to this repository.
